pkg/domain/infra/tunnel/containers.go

```diff
@@ -360,31 +360,35 @@ func (ic *ContainerEngine) ContainerCreate(ctx context.Context, s *specgen.SpecG
 func (ic *ContainerEngine) ContainerLogs(_ context.Context, nameOrIDs []string, options entities.ContainerLogsOptions) error {
 	since := options.Since.Format(time.RFC3339)
 	tail := strconv.FormatInt(options.Tail, 10)
-	stdout := options.Writer != nil
+	stdout := options.StdoutWriter != nil
+	stderr := options.StderrWriter != nil
 	opts := containers.LogOptions{
 		Follow:     &options.Follow,
 		Since:      &since,
-		Stderr:     &stdout,
+		Stderr:     &stderr,
 		Stdout:     &stdout,
 		Tail:       &tail,
 		Timestamps: &options.Timestamps,
 		Until:      nil,
 	}
 
 	var err error
-	outCh := make(chan string)
+	stdoutCh := make(chan string)
+	stderrCh := make(chan string)
 	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
-		err = containers.Logs(ic.ClientCxt, nameOrIDs[0], opts, outCh, outCh)
+		err = containers.Logs(ic.ClientCxt, nameOrIDs[0], opts, stdoutCh, stderrCh)
 		cancel()
 	}()
 
 	for {
 		select {
 		case <-ctx.Done():
 			return err
-		case line := <-outCh:
-			_, _ = io.WriteString(options.Writer, line+"\n")
+		case line := <-stdoutCh:
+			_, _ = io.WriteString(options.StdoutWriter, line+"\n")
```

Comment:
Is this safe if `options.StdoutWriter` is nil? Or do we need to check for that?

Reply:
Yes we need to check for that. Good catch